<launch>
	<!--rosbag play from bagpath/bagfile.bag-->
	<param name="use_sim_time" value="true" />
	<arg name="bagpath" default="/home/kakeru/catkin_ws/src/pclbags"/>
	<arg name="bagfile" default="kensyu.bag"/>
	<node pkg="rosbag" type="play" name="pcl_bag_player" args="--clock $(arg bagpath)/$(arg bagfile)" />
 	<node pkg="tf" type="static_transform_publisher" name="cloud_tf" args="0 0 1 0 0 3.14 hokuyo3d cloud_tf 100" />

	<node pkg="nodelet" type="nodelet" name="pcl_manager" args="manager" output="screen" />

	<!--Run a Voxel Grid filter for downsampling-->
	<node pkg="nodelet" type="nodelet" name="voxel_grid" args="load pcl/VoxelGrid pcl_manager" output="screen" >
		<remap from="~input" to="hokuyo3d/hokuyo_cloud2" />
		<remap from="~output" to="cloud_fd" />
		<rosparam>
filter_field_name: z
filter_limit_min: 0.05
filter_limit_max: 1.0
filter_limit_negative: False
leaf_size: 0.1</rosparam>
	</node>

    <!-- make ocupancy grid map -->
 	<node pkg="octomap_server" type="octomap_server_node" name="octomap_server">
 		<param name="resolution" value="0.15" />
 		<param name="frame_id" type="string" value="cloud_tf" />
 		<param name="sensor_model/max_range" value="10.0" />
 		<remap from="cloud_in" to="/cloud_fd" />
 	</node>

 	<node name="rviz" pkg="rviz" type="rviz" args="-d $(find pcl_something)/rviz_config/rviz_oct.rviz" />

</launch>